OpenEdXMobile/src/main/java/org/edx/mobile/view/CourseUnitOnlyOnYoutubeFragment.java

```diff
@@ -31,16 +31,22 @@ public void onCreate(Bundle savedInstanceState) {
     @Override
     public View onCreateView(LayoutInflater inflater, ViewGroup container,
                              Bundle savedInstanceState) {
-        View v = inflater.inflate(R.layout.fragment_course_unit_only_on_youtube, container, false);
-        ((TextView) v.findViewById(R.id.only_youtube_available_message)).setText(R.string.assessment_only_on_youtube);
-        v.findViewById(R.id.view_on_youtube_button).setOnClickListener(new View.OnClickListener() {
-            @Override
-            public void onClick(View v) {
-                Intent i = new Intent(Intent.ACTION_VIEW);
-                i.setData(Uri.parse(((VideoBlockModel) unit).getData().encodedVideos.youtube.url));
-                startActivity(i);
-            }
+        final View courseUnitOnlyOnYoutubeView = inflater.inflate(R.layout.fragment_course_unit_only_on_youtube, container, false);
+        if (environment.getConfig().getEmbeddedYoutubeConfig().isYoutubeEnabled()) {
+            ((TextView) courseUnitOnlyOnYoutubeView.findViewById(R.id.only_youtube_available_message)).setText(R.string.assessment_needed_updating_youtube);
+            courseUnitOnlyOnYoutubeView.findViewById(R.id.update_youtube_button).setVisibility(View.VISIBLE);
+            courseUnitOnlyOnYoutubeView.findViewById(R.id.update_youtube_button).setOnClickListener(v -> {
+                final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(AppConstants.BROWSER_PLAYSTORE_YOUTUBE_URI));
```

Comment:
```                
final Intent intent = new Intent(Intent.ACTION_VIEW, Uri.parse(AppConstants.BROWSER_PLAYSTORE_YOUTUBE_URI));
startActivity(intent);
```
this code is automatically redirected to the PlayStore app if installed, otherwise open the link in browser, so there is not need other line of code.

Reply:
> We should also make this kind of functions(in which intent is Intent.ACTION_VIEW) a utility function.

Use an already existing method of `BrowserUtil`